### PR TITLE
[fix](memory): avoid coredump when list pointer is null

### PR DESCRIPTION
**Pull request by levy5307**
# Proposed changes

Issue Number: [5664](https://github.com/apache/doris/issues/5664)

## Problem summary

- To avoid the coredump memtioned in above issue, adding the validation of pointer before using it.
BTW, It's a best practise to validate a pointer before using it.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ]  I don't know
3. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [x] No Need
5. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
7. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
9. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [[dev@doris.apache.org](mailto:dev@doris.apache.org)](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by dataroaring — CHANGES_REQUESTED** [review]
Could you  add a regression test?

File: be/src/util/likely_util.h [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#pragma once
+
+namespace doris {
+#define likely(pred) (__builtin_expect(!!(pred), 1))
```
Comment (Gabriel39) [inline be/src/util/likely_util.h:21]: We already have LIKELY and UNLIKELY in common/compiler_util.h. Maybe you do not need to add a new one

Comment (levy5307) [reply be/src/util/likely_util.h:21]: OK

**Comment by levy5307** [comment]
> Could you add a regression test?

Sure

**Comment by levy5307** [comment]
> Could you add a regression test?

It's hard to add a regression test because this bug is not a problem bound to arise. So I add a unit test for this function instead.

**Review by yiguolei — APPROVED** [review]
LGTM

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**